src/wp-includes/rest-api/endpoints/class-wp-rest-font-library-controller.php

```diff
@@ -0,0 +1,394 @@
+			if ( is_wp_error( $result ) ) {
+				return $result;
+			}
+		}
+
+		return new WP_REST_Response( __( 'Font family uninstalled successfully.' ), 200 );
+	}
+
+	/**
+	 * Checks whether the user has permissions to update the Font Library.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return true|WP_Error True if the request has write access for the item, WP_Error object otherwise.
+	 */
+	public function update_font_library_permissions_check() {
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return new WP_Error(
+				'rest_cannot_update_font_library',
+				__( 'Sorry, you are not allowed to update the Font Library on this site.' ),
+				array(
+					'status' => rest_authorization_required_code(),
+				)
+			);
+		}
+
+		// The update endpoints requires write access to the temp and the fonts directories.
+		$temp_dir   = get_temp_dir();
+		$upload_dir = wp_upload_dir()['basedir'];
+		if ( ! is_writable( $temp_dir ) || ! wp_is_writable( $upload_dir ) ) {
+			return new WP_Error(
+				'rest_cannot_write_fonts_folder',
+				__( 'Error: WordPress does not have permission to write the fonts folder on your server.' ),
+				array(
+					'status' => 500,
+				)
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Installs new fonts.
+	 *
+	 * Takes a request containing new fonts to install, downloads their assets, and adds them
+	 * to the Font Library.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param WP_REST_Request $request The request object containing the new fonts to install
+	 *                                 in the request parameters.
+	 * @return WP_REST_Response|WP_Error The updated Font Library post content.
+	 */
+	public function install_fonts( $request ) {
+		// Get new fonts to install.
+		$fonts_param = $request->get_param( 'fontFamilies' );
+
+		/*
+		 * As this is receiving form data, the font families are encoded as a string.
+		 * The form data is used  because local fonts need to use that format to
+		 * attach the files in the request.
+		 */
+		$fonts_to_install = json_decode( $fonts_param, true );
+
+		if ( empty( $fonts_to_install ) ) {
+			return new WP_Error(
+				'no_fonts_to_install',
+				__( 'No fonts to install' ),
+				array( 'status' => 400 )
+			);
+		}
+
+		// Get uploaded files (used when installing local fonts).
+		$files = $request->get_file_params();
+
+		// Iterates the fonts data received and creates a new WP_Font_Family object for each one.
+		$fonts_installed = array();
+		foreach ( $fonts_to_install as $font_data ) {
+			$font = new WP_Font_Family( $font_data );
+			$font->install( $files );
+			$fonts_installed[] = $font;
+		}
+
+		if ( empty( $fonts_installed ) ) {
+			return new WP_Error(
+				'error_installing_fonts',
+				__( 'Error installing fonts. No font was installed.' ),
+				array( 'status' => 500 )
+			);
+		}
+
+		$response = array();
+		foreach ( $fonts_installed as $font ) {
+			$response[] = $font->get_data();
+		}
+
+		return new WP_REST_Response( $response );
```

Comment:
```suggestion
		$data = array();
		foreach ( $fonts_installed as $font ) {
			$data[] = $font->get_data();
		}

		return rest_ensure_response( $data );
```

Reply:
We standardized the output of the endpoints here: https://github.com/WordPress/gutenberg/pull/54829

Reply:
Done in https://github.com/WordPress/wordpress-develop/commit/51164435b77deed771f8771c0e7bfb7535b91744 ✅

Reply:
As this is done, I think it's okay to mark this discussion as resolved, which will fold it. I don't think leaving it expanded adds helpful contextual information for contributors.